akka-http-core/src/main/scala/akka/http/scaladsl/model/Uri.scala

```diff
@@ -30,6 +30,14 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
   def isEmpty: Boolean
 
   /**
+   * Parses the given String into a Query instance.
+   */
+  def query(charset: Charset = UTF8, mode: Uri.ParsingMode = Uri.ParsingMode.Relaxed): Query = queryString match {
```

Comment:
Yes, good point, I forgot that relaxed parsing applies to path components, as well. I wouldn't get rid of the charset parameter, I'd just like to have the common case to be not more complicated than before.